Incompatible with nettle 4.0

Nettle 4.0 have API-break changes, specially in the `_digest()` functions, which now no longer takes the desired digest size as argument. More information [here](https://git.lysator.liu.se/nettle/nettle/-/blob/nettle_4.0_release_20260205/NEWS#L1-182).

It currently fails to build with Nettle 4.0 with the following errors:

```
In file included from /build/hefur/src/hefur/mimosa/mimosa/bittorrent/torrent-parser.cc:5:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Md2::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:52:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   52 |     MIMOSA_NETTLE_HASH(MD2, md2, Md2);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_md2_digest(md2_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:52:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   52 |     MIMOSA_NETTLE_HASH(MD2, md2, Md2);
      |     ^~~~~~~~~~~~~~~~~~
In file included from /build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:3:
/usr/include/nettle/md2.h:68:1: note: declared here
   68 | md2_digest(struct md2_ctx *ctx,
      | ^~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Md5::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:53:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   53 |     MIMOSA_NETTLE_HASH(MD5, md5, Md5);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_md5_digest(md5_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:53:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   53 |     MIMOSA_NETTLE_HASH(MD5, md5, Md5);
      |     ^~~~~~~~~~~~~~~~~~
In file included from /build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:4:
/usr/include/nettle/md5.h:72:1: note: declared here
   72 | md5_digest(struct md5_ctx *ctx,
      | ^~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha1::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:54:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   54 |     MIMOSA_NETTLE_HASH(SHA1, sha1, Sha1);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha1_digest(sha1_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:54:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   54 |     MIMOSA_NETTLE_HASH(SHA1, sha1, Sha1);
      |     ^~~~~~~~~~~~~~~~~~
In file included from /build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:5:
/usr/include/nettle/sha1.h:74:1: note: declared here
   74 | sha1_digest(struct sha1_ctx *ctx,
      | ^~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha224::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:55:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   55 |     MIMOSA_NETTLE_HASH(SHA224, sha224, Sha224);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha224_digest(sha256_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:55:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   55 |     MIMOSA_NETTLE_HASH(SHA224, sha224, Sha224);
      |     ^~~~~~~~~~~~~~~~~~
In file included from /build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:6:
/usr/include/nettle/sha2.h:104:1: note: declared here
  104 | sha224_digest(struct sha256_ctx *ctx,
      | ^~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha256::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:56:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   56 |     MIMOSA_NETTLE_HASH(SHA256, sha256, Sha256);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha256_digest(sha256_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:56:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   56 |     MIMOSA_NETTLE_HASH(SHA256, sha256, Sha256);
      |     ^~~~~~~~~~~~~~~~~~
/usr/include/nettle/sha2.h:86:1: note: declared here
   86 | sha256_digest(struct sha256_ctx *ctx,
      | ^~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha384::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:57:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   57 |     MIMOSA_NETTLE_HASH(SHA384, sha384, Sha384);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha384_digest(sha512_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:57:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   57 |     MIMOSA_NETTLE_HASH(SHA384, sha384, Sha384);
      |     ^~~~~~~~~~~~~~~~~~
/usr/include/nettle/sha2.h:151:1: note: declared here
  151 | sha384_digest(struct sha512_ctx *ctx,
      | ^~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha512::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:58:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   58 |     MIMOSA_NETTLE_HASH(SHA512, sha512, Sha512);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha512_digest(sha512_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:58:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   58 |     MIMOSA_NETTLE_HASH(SHA512, sha512, Sha512);
      |     ^~~~~~~~~~~~~~~~~~
/usr/include/nettle/sha2.h:133:1: note: declared here
  133 | sha512_digest(struct sha512_ctx *ctx,
      | ^~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha3_224::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:59:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   59 |     MIMOSA_NETTLE_HASH(SHA3_224, sha3_224, Sha3_224);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha3_224_digest(sha3_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:59:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   59 |     MIMOSA_NETTLE_HASH(SHA3_224, sha3_224, Sha3_224);
      |     ^~~~~~~~~~~~~~~~~~
In file included from /build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:7:
/usr/include/nettle/sha3.h:126:1: note: declared here
  126 | sha3_224_digest(struct sha3_ctx *ctx, uint8_t *digest);
      | ^~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha3_256::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:60:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   60 |     MIMOSA_NETTLE_HASH(SHA3_256, sha3_256, Sha3_256);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha3_256_digest(sha3_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:60:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   60 |     MIMOSA_NETTLE_HASH(SHA3_256, sha3_256, Sha3_256);
      |     ^~~~~~~~~~~~~~~~~~
/usr/include/nettle/sha3.h:132:1: note: declared here
  132 | sha3_256_digest(struct sha3_ctx *ctx, uint8_t *digest);
      | ^~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha3_384::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:61:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   61 |     MIMOSA_NETTLE_HASH(SHA3_384, sha3_384, Sha3_384);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha3_384_digest(sha3_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:61:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   61 |     MIMOSA_NETTLE_HASH(SHA3_384, sha3_384, Sha3_384);
      |     ^~~~~~~~~~~~~~~~~~
/usr/include/nettle/sha3.h:148:1: note: declared here
  148 | sha3_384_digest(struct sha3_ctx *ctx, uint8_t *digest);
      | ^~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh: In member function ‘char* mimosa::stream::Sha3_512::digest()’:
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:30: error: invalid conversion from ‘long unsigned int’ to ‘uint8_t*’ {aka ‘unsigned char*’} [-fpermissive]
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                              ^~~~~~~~~~~~~~~~
      |                              |
      |                              long unsigned int
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:62:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   62 |     MIMOSA_NETTLE_HASH(SHA3_512, sha3_512, Sha3_512);
      |     ^~~~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:41:22: error: too many arguments to function ‘void nettle_sha3_512_digest(sha3_ctx*, uint8_t*)’
   41 |         hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
      |                      ^
/build/hefur/src/hefur/mimosa/mimosa/bittorrent/../stream/hash.hh:62:5: note: in expansion of macro ‘MIMOSA_NETTLE_HASH’
   62 |     MIMOSA_NETTLE_HASH(SHA3_512, sha3_512, Sha3_512);
      |     ^~~~~~~~~~~~~~~~~~
/usr/include/nettle/sha3.h:154:1: note: declared here
  154 | sha3_512_digest(struct sha3_ctx *ctx, uint8_t *digest);
      | ^~~~~~~~~~~~~~~
[ 76%] Built target mimosa-tpl
make[2]: *** [mimosa/mimosa/bittorrent/CMakeFiles/mimosa-bittorrent.dir/build.make:93: mimosa/mimosa/bittorrent/CMakeFiles/mimosa-bittorrent.dir/torrent-parser.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:2130: mimosa/mimosa/bittorrent/CMakeFiles/mimosa-bittorrent.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/build/hefur/src/hefur/mimosa/mimosa/http/method-handler.cc: In member function ‘virtual bool mimosa::http::MethodHandler::options(mimosa::http::RequestReader&, mimosa::http::ResponseWriter&) const’:
/build/hefur/src/hefur/mimosa/mimosa/http/method-handler.cc:72:44: warning: unused parameter ‘request’ [-Wunused-parameter]
   72 |     MethodHandler::options(RequestReader & request,
      |                            ~~~~~~~~~~~~~~~~^~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/http/request-lexer.cc:9220:16: warning: ‘int yy_top_state(yyscan_t)’ defined but not used [-Wunused-function]
 9220 | 
      |                ^           
/build/hefur/src/hefur/mimosa/mimosa/http/request-lexer.cc:9211:17: warning: ‘void yy_pop_state(yyscan_t)’ defined but not used [-Wunused-function]
 9211 | 
      |                 ^           
/build/hefur/src/hefur/mimosa/mimosa/http/request-lexer.cc:9185:17: warning: ‘void yy_push_state(int, yyscan_t)’ defined but not used [-Wunused-function]
 9185 | 
      |                 ^            
/build/hefur/src/hefur/mimosa/mimosa/http/request-lexer.cc:8712:17: warning: ‘void yyunput(int, char*, yyscan_t)’ defined but not used [-Wunused-function]
 8712 | 
      |                 ^      
[ 78%] Linking CXX static library libmimosa-http.a
[ 78%] Built target mimosa-http
/build/hefur/src/hefur/mimosa/mimosa/rpc/channel.cc: In member function ‘void mimosa::rpc::Channel::sendCall(const mimosa::RefCountable<mimosa::rpc::BasicCall>::Ptr&)’:
/build/hefur/src/hefur/mimosa/mimosa/rpc/channel.cc:106:39: warning: ignoring return value of ‘bool google::protobuf::MessageLite::SerializeToArray(void*, int) const’, declared with attribute ‘nodiscard’ [-Wunused-result]
  106 |       call->request_->SerializeToArray(msg->rq_, rq_size);
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
In file included from /build/hefur/src/protobuf-34.1/src/google/protobuf/generated_enum_util.h:20,
                 from /build/hefur/src/protobuf-34.1/src/google/protobuf/extension_set.h:39,
                 from /build/hefur/src/protobuf-34.1/src/google/protobuf/descriptor.h:56,
                 from /build/hefur/src/protobuf-34.1/src/google/protobuf/message.h:113,
                 from /build/hefur/src/hefur/mimosa/mimosa/rpc/basic-call.hh:3,
                 from /build/hefur/src/hefur/mimosa/mimosa/rpc/service.hh:6,
                 from /build/hefur/src/hefur/mimosa/mimosa/rpc/service-map.hh:7,
                 from /build/hefur/src/hefur/mimosa/mimosa/rpc/channel.hh:12,
                 from /build/hefur/src/hefur/mimosa/mimosa/rpc/channel.cc:8:
/build/hefur/src/protobuf-34.1/src/google/protobuf/message_lite.h:767:44: note: declared here
  767 |   PROTOBUF_FUTURE_ADD_EARLY_NODISCARD bool SerializeToArray(void* data,
      |                                            ^~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/rpc/channel.cc: In member function ‘void mimosa::rpc::Channel::sendResponse(const mimosa::RefCountable<mimosa::rpc::BasicCall>::Ptr&)’:
/build/hefur/src/hefur/mimosa/mimosa/rpc/channel.cc:137:40: warning: ignoring return value of ‘bool google::protobuf::MessageLite::SerializeToArray(void*, int) const’, declared with attribute ‘nodiscard’ [-Wunused-result]
  137 |       call->response_->SerializeToArray(msg->rp_, rp_size);
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
/build/hefur/src/protobuf-34.1/src/google/protobuf/message_lite.h:767:44: note: declared here
  767 |   PROTOBUF_FUTURE_ADD_EARLY_NODISCARD bool SerializeToArray(void* data,
      |                                            ^~~~~~~~~~~~~~~~
/build/hefur/src/hefur/mimosa/mimosa/rpc/channel.cc: In member function ‘bool mimosa::rpc::Channel::handleResult()’:
/build/hefur/src/hefur/mimosa/mimosa/rpc/channel.cc:263:38: warning: ignoring return value of ‘bool google::protobuf::MessageLite::ParseFromArray(const void*, int)’, declared with attribute ‘nodiscard’ [-Wunused-result]
  263 |       call->response_->ParseFromArray(data, msg.rp_size_);
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/build/hefur/src/protobuf-34.1/src/google/protobuf/message_lite.h:691:3: note: declared here
  691 |   ParseFromArray(const void* data, int size);
      |   ^~~~~~~~~~~~~~
[ 78%] Linking CXX static library libmimosa-rpc.a
[ 78%] Built target mimosa-rpc
[ 78%] Linking CXX executable protoc-gen-mimosa
[ 78%] Built target protoc-gen-mimosa
make: *** [Makefile:146: all] Error 2
```

System information:
  - OS: Arch Linux (rolling)
  - Compiler: GCC 15.2
  - Nettle: 4.0

--- a/mimosa/stream/hash.hh
+++ b/mimosa/stream/hash.hh
@@ -5,6 +5,7 @@
 # include <nettle/sha1.h>
 # include <nettle/sha2.h>
 # include <nettle/sha3.h>
+# include <nettle/version.h>
 
 # include "stream.hh"
 
@@ -12,6 +13,12 @@ namespace mimosa
 {
   namespace stream
   {
+
+#if NETTLE_VERSION_MAJOR >= 4
+#define MIMOSA_NETTLE_DIGEST_FUNC(hash, ctx_, digest_) hash##_digest(&ctx_, (uint8_t *)digest_)
+#else
+#define MIMOSA_NETTLE_DIGEST_FUNC(hash, ctx_, digest_) hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_)
+#endif
 
 #define MIMOSA_NETTLE_HASH(HASH, hash, Class)                           \
     class Class : public Stream                                         \
@@ -38,7 +45,7 @@ namespace mimosa
                                                                         \
       char * digest()                                                   \
       {                                                                 \
-        hash##_digest(&ctx_, sizeof (digest_), (uint8_t *)digest_);     \
+        MIMOSA_NETTLE_DIGEST_FUNC(hash, ctx_, digest_);                 \
         return digest_;                                                 \
       }                                                                 \
                                                                         \
@@ -61,6 +68,7 @@ namespace mimosa
     MIMOSA_NETTLE_HASH(SHA3_384, sha3_384, Sha3_384);
     MIMOSA_NETTLE_HASH(SHA3_512, sha3_512, Sha3_512);
 
+#undef MIMOSA_NETTLE_DIGEST_FUNC
 #undef MIMOSA_NETTLE_HASH
   }
   }

